app/src/main/java/com/therap/amin/currencyconverter/CurrencyValueSaver.java

```diff
@@ -20,71 +21,75 @@
+        inputCurrencySpinner = (Spinner) findViewById(R.id.spnLeftCurrency);
+        outputCurrencySpinner = (Spinner) findViewById(R.id.spnRightCurrency);
+        saveButton = (Button) findViewById(R.id.btnSave);
+        etConversionValue = (EditText) findViewById(R.id.etRightCurrencyValue);
 
+        sharedPreferences = getSharedPreferences(Constants.preferenceKey, Context.MODE_PRIVATE);
 
-        leftCurrency = (Spinner) findViewById(R.id.spnLeftCurrency);
-        rightCurrency = (Spinner) findViewById(R.id.spnRightCurrency);
+        ArrayAdapter<String> inputCurrencyAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, availableCurrencies);
+        inputCurrencySpinner.setAdapter(inputCurrencyAdapter);
 
-        save = (Button) findViewById(R.id.btnSave);
+        ArrayAdapter<String> outputCurrencyAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, availableCurrencies);
+        outputCurrencySpinner.setAdapter(outputCurrencyAdapter);
 
-        conversionValue = (EditText) findViewById(R.id.etRightCurrencyValue);
-
-        sharedPreferences = getSharedPreferences("currencyPreference", Context.MODE_PRIVATE);
-
-
-        ArrayAdapter<String> leftCurrencyAdapter = new ArrayAdapter<String>(this,android.R.layout.simple_spinner_item,currencies);
-        leftCurrency.setAdapter(leftCurrencyAdapter);
-
-        ArrayAdapter<String> rightCurrencyAdapter = new ArrayAdapter<String>(this,android.R.layout.simple_spinner_item,currencies);
-        rightCurrency.setAdapter(rightCurrencyAdapter);
-
-
-
-        save.setOnClickListener(new View.OnClickListener() {
+        saveButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
 
-                if (!conversionValue.getText().toString().equals("")) {
-                    SharedPreferences.Editor editor = sharedPreferences.edit();
-                    String left = leftCurrency.getSelectedItem().toString();
-                    String right =rightCurrency.getSelectedItem().toString();
-                    if (left.equals(right)) {
-                        Toast.makeText(getApplicationContext(),"You can't set "+left+" -> "+right+" conversion value",Toast.LENGTH_SHORT).show();
-                        conversionValue.getText().clear();
+                if (!etConversionValue.getText().toString().equals("")) {
+
+                    String inputCurrency = inputCurrencySpinner.getSelectedItem().toString();
+                    String outputCurrency = outputCurrencySpinner.getSelectedItem().toString();
+                    if (inputCurrency.equals(outputCurrency)) {
+                        Toast.makeText(getApplicationContext(), "You can't set " + inputCurrency + " -> " + outputCurrency + " conversion value", Toast.LENGTH_SHORT).show();
+                    } else if (sharedPreferences.contains(Constants.inputCurrencyKey)) {
```

Comment:
you can merge this two condition checking easily by adding ``sharedPreferences.contains(Constants.inputCurrencyKey)`` checking at **addToPreference(..)** method.